### DISCUSSION TITLE
Renovate updates Docker Image to non existing digest

### DISCUSSION THREAD

**Discussion by Skaronator** [renovatebot/renovate#27469] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### Describe the problem

Hi,

Sometimes Renovate updates to a non-existing Docker Digest. It already happened 2-3 times and it's mostly the mariadb and redis image.

Here is the PR that updates to a not working Docker Digest: https://github.com/Skaronator/homelab/pull/852
When looking up that exist digest, I can see that it exists in Docker Hub, but it's empty: https://hub.docker.com/layers/library/mariadb/11.3.2/images/sha256-eb37f58646a901dc7727cf448cae36daaefaba79de33b5058dab79aa4c04aefb?context=explore

```bash
$ docker pull mariadb:11.3.2@sha256:eb37f58646a901dc7727cf448cae36daaefaba79de33b5058dab79aa4c04aefb
docker.io/library/mariadb@sha256:eb37f58646a901dc7727cf448cae36daaefaba79de33b5058dab79aa4c04aefb: Pulling from library/mariadb
no matching manifest for linux/amd64 in the manifest list entries
```

I reverted the Renovate PR here: https://github.com/Skaronator/homelab/pull/853

And after a while, Renovate created a new PR but this time with a new digest: https://github.com/Skaronator/homelab/pull/854
The new digest works as expected:
```bash
$ docker pull mariadb:11.3.2@sha256:c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3
docker.io/library/mariadb@sha256:c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3: Pulling from library/mariadb
Digest: sha256:c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3
Status: Image is up to date for mariadb@sha256:c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3
docker.io/library/mariadb:11.3.2@sha256:c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3
```
(Already pulled)

But weirdly, the Docker Hub is also empty: https://hub.docker.com/layers/library/mariadb/11.3.2/images/sha256-c33b9fe0c4c10e2a94b62ba952e006c185dfece517c7e3303eff8473f2f523f3?context=explore

Reproducing this is almost impossible since the second renovate run always works as expected, and we can only test this the moment a new update is available.

### Relevant debug logs

Full Logs: https://gist.github.com/Skaronator/f6c794ec5e89a9373cbe37dcb916aa81
<details><summary>Logs</summary>

```json
          {
            "depName": "mariadb",
            "currentValue": "11.2.3",
            "currentDigest": "sha256:55d9608d84658cbc691f52cd6389386a15e6c671cb5708b9f3af5782af3b4dbb",
            "datasource": "docker",
            "replaceString": "11.2.3@sha256:55d9608d84658cbc691f52cd6389386a15e6c671cb5708b9f3af5782af3b4dbb",
            "versioning": "docker",
            "autoReplaceStringTemplate": "{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "11.3.2",
                "newValue": "11.3.2",
                "newDigest": "sha256:eb37f58646a901dc7727cf448cae36daaefaba79de33b5058dab79aa4c04aefb",
                "releaseTimestamp": "2024-02-21T01:36:06.309Z",
                "newMajor": 11,
                "newMinor": 3,
                "updateType": "minor",
                "branchName": "renovate/mariadb-11.x"
              }
            ],
            "packageName": "mariadb",
            "warnings": [],
            "registryUrl": "https://index.docker.io",
            "currentVersion": "11.2.3",
            "isSingleVersion": true,
            "fixedVersion": "11.2.3"
          },
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Please check the log from when the PR was created and look for all instances of that SHA. This is usually caused when people publish images to Docker Hub incrementally (platform by platform) and Renovate queries when only a subset of platforms are there

**Reply by Skaronator** [reply]
Ah yeah, this makes sense. I've linked the full debug log above, but it shows only that single SHA.

Do you have any suggestions to avoid this situation?

**Reply by rarkins** [reply, marked as answer]
Setting minimumReleaseAge should be a workaround for anything on docker hub

**Reply by viceice** [reply]
I'm working on a refactoring on the digest handling, but it needs some more time to get it done.

**Reply by Skaronator** [reply]
Ah great. I thought the minimumReleaseAge only accepted "x days" and not just "x hours". Using this now:

```json
  "packageRules": [
    {
      "description": "Wait for stable release. Updating this instantly will cause issues since amd64 builds always take a tiny bit more",
      "matchPackageNames": ["redis", "mariadb", "postgres"],
      "minimumReleaseAge": "4 hours"
    }
  ]
```
